docs/sources/alerting/rules.md

```diff
@@ -51,7 +51,9 @@ Here you can specify the name of the alert rule and how often the scheduler shou
 
 ### For
 
-> This setting is available in Grafana 5.4 and above.
+> **Important note regarding No Data:**
+>
+> Do not use `For` with the `If no data or all values are null` setting set to `No Data`. The triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.
```

Comment:
Is this a temporary notice? Shouldn't this be fixed in the next versions?

Reply:
Given that we're not sure if we want to add this support and/or how to implement it given next generation alerting is going to happen eventually and/or if we have time I think documenting this is needed until we've figured out what to do with this.